mesonbuild/dependencies/ui.py

```diff
@@ -192,8 +192,39 @@ def __init__(self, name: str, environment: 'Environment', kwargs: DependencyObje
                 lib_name = 'vulkan-1'
                 lib_dir = 'Lib32'
                 inc_dir = 'Include'
-                if detect_cpu_family(self.env.coredata.compilers.host) == 'x86_64':
-                    lib_dir = 'Lib'
+                build_cpu = self.env.machines.build.cpu_family
+                host_cpu = self.env.machines.host.cpu_family
+                unsupported_target_arch = False
+                if build_cpu == 'x86_64':
+                    if host_cpu == build_cpu:
+                        lib_dir = 'Lib'
+                    elif host_cpu == 'aarch64':
+                        lib_dir = 'Lib-ARM64'
+                    elif host_cpu == 'x86':
+                        lib_dir = 'Lib32'
+                    else:
+                        unsupported_target_arch = True
+                elif build_cpu == 'aarch64':
+                    if host_cpu == build_cpu:
+                        lib_dir = 'Lib'
+                    elif host_cpu == 'x86_64':
+                        lib_dir = 'Lib-x64'
+                    elif host_cpu == 'x86':
+                        lib_dir = 'Lib32'
+                    else:
+                        unsupported_target_arch = True
+                elif build_cpu == 'x86':
+                    if host_cpu == build_cpu:
+                        lib_dir = 'Lib32'
+                    if host_cpu == 'aarch64':
+                        lib_dir = 'Lib-ARM64'
+                    elif host_cpu == 'x86_64':
+                        lib_dir = 'Lib'
+                    else:
+                        unsupported_target_arch = True
+
+                if unsupported_target_arch:
+                    raise DependencyException('Target architecture \'%s\' is not supported for this Vulkan SDK.' % host_cpu)
```

Comment:
```suggestion
                    raise DependencyException(f'Target architecture \'{host_cpu}\' is not supported for this Vulkan SDK.')
```

We prefer f-strings for simple substitutions or `str.format()` for complex ones.

Ah, you thought of more error cases than I did, I only noticed the `build_cpu` being unsupported I think it would be simpler to set `lib_dir` to some initial sentinel value like` lib_dir = ''` and then check for that at the end for the error condition.